templates/al2023/provisioners/install-nvidia-driver.sh

```diff
@@ -0,0 +1,115 @@
+
+sudo sed -i 's/gpgcheck=0/gpgcheck=1/g' /etc/yum.repos.d/nvidia-container-toolkit.repo /etc/yum.repos.d/cuda-amzn2023.repo 
+
+################################################################################
+### Install drivers ############################################################
+################################################################################
+sudo mkdir -p /etc/dkms
+echo "MAKE[0]=\"'make' -j$(grep -c processor /proc/cpuinfo) module\"" | sudo tee /etc/dkms/nvidia.conf
+sudo dnf -y install kernel-modules-extra.x86_64
+
+function archive-open-kmods(){
+  sudo dnf --setopt=install_weak_deps=False -y module install nvidia-driver:${NVIDIA_MAJOR_DRIVER_VERSION}-open
+  # The DKMS package name differs between the RPM and the dkms.conf in the OSS kmod sources
+  # TODO: can be removed if this is merged: https://github.com/NVIDIA/open-gpu-kernel-modules/pull/567
+  sudo sed -i 's/PACKAGE_NAME="nvidia"/PACKAGE_NAME="nvidia-open"/g' /var/lib/dkms/nvidia-open/$(kmod-util module-version nvidia-open)/source/dkms.conf
+
+  sudo kmod-util archive nvidia-open
+
+  KMOD_MAJOR_VERSION=$(sudo kmod-util module-version nvidia-open | cut -d. -f1)
+  SUPPORTED_DEVICE_FILE="${WORKING_DIR}/gpu/nvidia-open-supported-devices-${KMOD_MAJOR_VERSION}.txt"
+  sudo mv "${SUPPORTED_DEVICE_FILE}" /etc/eks/
+
+  sudo kmod-util remove nvidia-open
+
+  sudo dnf -y module remove --all nvidia-driver
+  sudo dnf -y module reset nvidia-driver
+} 
+
+function archive-proprietary-kmod(){
+  sudo dnf --setopt=install_weak_deps=False -y module install nvidia-driver:${NVIDIA_MAJOR_DRIVER_VERSION}-dkms
+  sudo kmod-util archive nvidia
+  sudo kmod-util remove nvidia
+}
+
+archive-open-kmods
+archive-proprietary-kmod
+
+################################################################################
+### Prepare for nvidia bootstrap ###############################################
+################################################################################
+
+sudo mv ${WORKING_DIR}/gpu/nvidia-kmod-load.sh /etc/eks/
+sudo mv ${WORKING_DIR}/gpu/initialize_nvidia_clock.sh /etc/eks/
+sudo mv ${WORKING_DIR}/gpu/bootstrap_gpu.sh /etc/eks/
+sudo mv ${WORKING_DIR}/gpu/bootstrap_gpu.service /etc/systemd/system/bootstrap_gpu.service
+sudo systemctl daemon-reload
+sudo systemctl enable bootstrap_gpu
+
+################################################################################
+### Install Cuda ###############################################################
+################################################################################
+
+declare -A NVIDIA_DRIVER_CUDA_VERSION_MAP=(["555"]="12-5" ["560"]="12-6")
+CUDA_VERSION=${NVIDIA_DRIVER_CUDA_VERSION_MAP[${NVIDIA_MAJOR_DRIVER_VERSION}]}
+
+if [ -z $CUDA_VERSION ]; then
+  echo "undefined CUDA version for Nvidia driver version: ${NVIDIA_MAJOR_DRIVER_VERSION}"
+  exit 1
+fi
+
+sudo dnf --setopt=install_weak_deps=False -y install cuda-toolkit-${CUDA_VERSION} cuda-*-${CUDA_VERSION}-*
```

Comment:
were you going to remove this? the toolkit can go, at least

Reply:
yes, I'm dropping CUDA.